internal/signer/v2/v2_test.go

```diff
@@ -0,0 +1,131 @@
+package v2
+
+import (
+	"bytes"
+	"net/http"
+	"net/url"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/service"
+	"github.com/stretchr/testify/assert"
+)
+
+func buildSigner(serviceName string, region string, signTime time.Time, query url.Values) signer {
+	endpoint := "https://" + serviceName + "." + region + ".amazonaws.com"
+	body := []byte(query.Encode())
+	reader := bytes.NewReader(body)
+	req, _ := http.NewRequest("POST", endpoint, reader)
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Add("Content-Length", string(len(body)))
+
+	signer := signer{
+		Request: req,
+		Time:    signTime,
+		Credentials: credentials.NewStaticCredentials(
+			"AKIAJG6V72ZBDMWPMSWQ",
+			"P9MvpCRxpwo2UexwMYbduHEoVcBPJQZO2GVsKNCD",
```

Comment:
Agreed. Interestingly enough, I could not find any test suites for V2 signature, so I captured a SimpleDB request using fiddler and used that as the basis for this test. Of course I invalidated the credentials after doing this. I take your point though. It is difficult to check that these credentials are no longer valid and it is better to use mock credentials.